src/conn/conn_prefetch.c

```diff
@@ -157,6 +157,39 @@ __wt_conn_prefetch_queue_push(WT_SESSION_IMPL *session, WT_REF *ref)
     return (ret);
 }
 
+/*
+ * __wt_conn_prefetch_clear_tree --
+ *     Clear pages from the pre-fetch queue, either all pages on the queue or pages from the current
+ *     btree - depending on input parameters.
+ */
+int
+__wt_conn_prefetch_clear_tree(WT_SESSION_IMPL *session, bool all)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DATA_HANDLE *dhandle;
+    WT_PREFETCH_QUEUE_ENTRY *pe, *pe_tmp;
+
+    conn = S2C(session);
+    dhandle = session->dhandle;
+
+    WT_ASSERT_ALWAYS(session, all || dhandle != NULL,
+      "Pre-fetch needs to save a valid dhandle when clearing the queue for a btree");
+
+    __wt_spin_lock(session, &conn->prefetch_lock);
+    TAILQ_FOREACH_SAFE(pe, &conn->pfqh, q, pe_tmp)
+    {
+        if (all || pe->dhandle == dhandle) {
+            TAILQ_REMOVE(&conn->pfqh, pe, q);
+            F_CLR(pe->ref, WT_REF_FLAG_PREFETCH);
+            __wt_free(session, pe);
+            --conn->prefetch_queue_count;
```

Comment:
Can you add the following assertion:
```
WT_ASSERT(session, conn->prefetch_queue_count > 0):
```

Reply:
I understand the intent of this but I think it would be more intuitive to assert/check that the queue count is zero after we've finished looping through the entries.